src/components/pipelines-overview/dateTime.ts

```diff
@@ -97,6 +97,49 @@ export const formatTime = (time: string): string => {
   return timestring;
 };
 
+export const formatTimeLastRunTime = (time: string): string => {
+  let timeValue = time?.split(/\s+/);
+  if (timeValue === undefined) {
+    return '-';
+  }
+  if(timeValue.length > 1){
+    // Check for the presence of each component
+    const hasYear = timeValue.includes('year') || timeValue.includes('years');
+    const hasMonth = timeValue.includes('month') || timeValue.includes('months');
+    const hasDay = timeValue.includes('day') || timeValue.includes('days');
+
+    // Return the most significant time component
+    if (hasYear) {
+      return timeValue.includes('year') ? timeValue[timeValue.indexOf('year') - 1] + ' year ago' : timeValue[timeValue.indexOf('years') - 1] + ' years ago';
+    } else if (hasMonth) {
+      return timeValue.includes('month') ? timeValue[timeValue.indexOf('month') - 1] + ' month ago' : timeValue[timeValue.indexOf('months') - 1] + ' months ago';
+    } else if (hasDay) {
+      return timeValue.includes('day') ? timeValue[timeValue.indexOf('day') - 1] + ' day ago' : timeValue[timeValue.indexOf('days') - 1] + ' days ago';
+    }
+    else {
+      return `${timeValue.pop()} ago`;
+    }
+  }
+  else{
+    const t = time?.split(/[:.]+/);
+    if (t == undefined) {
+      return '-';
+    }
+
+    let timestring = '';
+    if (t[0] != '00') {
+      timestring = t[0].replace(/^0+/, '') + ' hour ago';
+    }
+    else if (t[1] != '00') {
+      timestring = t[1].replace(/^0+/, '') + ' min ago';
+    }
+    else if (t[2] != '00') {
+      timestring = t[2].replace(/^0+/, '') + ' sec ago';
+    }
+    return `${timestring}`;
+  }
+};
+
```

Comment:
Small nit: we can add logic for adding plural case - egs- 2 month**s** ago instead of 2 month ago

Reply:
It is already added. Check formatTimeLastRunTime method.

Reply:
Yes, it is added for months, weeks and years. Not for hour, sec and min. Updated the comment